lua/xcodebuild/platform/macos.lua

```diff
@@ -24,8 +26,20 @@ function M.launch_app(appPath, callback)
     end
   end
 
+  local function write_logs(_, output)
+    if output[#output] == "" then
+      table.remove(output, #output)
+    end
+    appdata.append_app_logs(output)
+  end
+
+  appdata.clear_app_logs()
+
   return vim.fn.jobstart(command, {
     env = appdata.read_env_vars(),
+    pty = true,
+    on_stdout = write_logs,
+    on_stderr = write_logs,
     on_exit = function(_, code)
```

Comment:
The thing I liked about this functionality was that I was able to run my debug app from Neovim and keep it running, because `open` command exits immediately. This approach binds the app process with Neovim process. If I kill Neovim, my app will be killed as well.

Also, this is how it works for other platforms (`detach = true` is set) for `jobstart`. Unfortunately, `detach` doesn't work with `pty = true`. Maybe there is another way?

If not, maybe you could add command for macOS apps called "Run without building (detached)"?

Reply:
To add a new command you can check these commits to see which places should be updated:
https://github.com/wojciech-kulik/xcodebuild.nvim/commit/4c608d84526d3cd54cd0d304ef918074bec103e1
https://github.com/wojciech-kulik/xcodebuild.nvim/commit/1357a3e0e125c2cbd140b6732b2806ad34035608

Reply:
No, I don't think that's possible. Using `pty = true` in `vim.fn.jobstart` creates a pseudo-terminal for the process, which ties it to the terminal session. Detaching from the process while using a PTY is not directly supported because the process remains connected to the terminal.

I can add a new command to run the app in detached mode. Would you like me to change the default behavior so that the regular build and run actions stay attached to the process, even on iOS? This would ensure consistency across iOS and macOS while aligning the behavior more closely with Xcode.

Reply:
> Would you like me to change the default behavior so that the regular build and run actions stay attached to the process, even on iOS?

I can't decide on that :D, but I would prefer to keep the old behavior as it is. It won't be consistent, but maybe just for macOS let's do it as you proposed, without any extra commands for now.